src/officehours_api/serializers.py

```diff
@@ -83,6 +88,39 @@ class Meta:
         fields = ['id', 'username', 'first_name', 'last_name']
 
 
+class QueueAttendeeSerializer(serializers.ModelSerializer):
+    '''
+    Serializer used when viewing queue as an attendee.
+    '''
+    context: UserContext
+
+    hosts = NestedUserSerializer(many=True, read_only=True)
+    line_length = serializers.SerializerMethodField(read_only=True)
+    my_meeting = serializers.SerializerMethodField(read_only=True)
+    allowed_backends = serializers.ListField(child=serializers.CharField())
+
+    class Meta:
+        model = Queue
+        fields = ['id', 'name', 'created_at', 'description', 'hosts', 'line_length', 'my_meeting', 'status',
+                  'allowed_backends', 'inperson_location']
+
+    @extend_schema_field(serializers.IntegerField)
+    def get_line_length(self, obj):
+        return len([meeting for meeting in obj.meeting_set.all() if meeting.status != MeetingStatus.STARTED])
+
+    @extend_schema_field(NestedMyMeetingSerializer)
+    def get_my_meeting(self, obj):
+        user = self.context['user']
+        my_meeting = (
+            obj.meeting_set.filter(attendees__in=[self.context['user']]).first()
+            if user.is_authenticated else None
+        )
+        if not my_meeting:
+            return None
+        serializer = NestedMyMeetingSerializer(my_meeting, context=self.context)
+        return serializer.data
```

Comment:
This was moved because it was referenced in `MyUserSerializer` before it was defined.